From a45fdcfa2d219e986ae30377109d1c6771937629 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Wed, 22 Feb 2012 14:12:37 -0200
Subject: [RHEL6 qemu-kvm PATCH 2/3] scsi: fix fw path

RH-Author: Paolo Bonzini <pbonzini@redhat.com>
Message-id: <1329919979-20948-81-git-send-email-pbonzini@redhat.com>
Patchwork-id: 37560
O-Subject: [RHEL 6.3 qemu-kvm PATCH v2 080/102] scsi: fix fw path
Bugzilla: 782029
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>

The pre-1.0 firmware path for SCSI devices already included the LUN
using the suffix argument to add_boot_device_path.  Avoid that it is
included twice, and convert the colons to commas for consistency with
other kinds of devices

[ The previous patch was included in Kevin's pull request at the same
  time Andreas Faerber (who actually knows OpenFirmware) was reviewing it
  and I was sending v2.  In short, this is the interdiff between v1 and v2. ]

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from 795928f61daff10b06bb21e837bc5bf0227cf076)
---
 hw/scsi-bus.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 hw/scsi-bus.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hw/scsi-bus.c b/hw/scsi-bus.c
index 49c1bb7..326b1f6 100644
--- a/hw/scsi-bus.c
+++ b/hw/scsi-bus.c
@@ -1448,8 +1448,8 @@ static char *scsibus_get_fw_dev_path(DeviceState *dev)
     SCSIDevice *d = DO_UPCAST(SCSIDevice, qdev, dev);
     char path[100];
 
-    snprintf(path, sizeof(path), "%s@%d,%d,%d", qdev_fw_name(dev),
-             d->channel, d->id, d->lun);
+    snprintf(path, sizeof(path), "channel@%x/%s@%x,%x", d->channel,
+             qdev_fw_name(dev), d->id, d->lun);
 
     return strdup(path);
 }
-- 
1.7.3.2

